docs/madr/decisions/020-kds-redesign.md

```diff
@@ -0,0 +1,179 @@
+# KDS Redesign
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4354
+
+## Context and Problem Statement
+
+The current implementation of KDS is quite complex, and there may be some bugs, such as the one found at https://github.com/kumahq/kuma/pull/5373, which was very difficult to identify and resolve. The use of a bi-directional stream for configuration exchange between the zone and the global can also be challenging to implement. Additionally, KDS operates as a State of The World, meaning that whenever an item in the resource type changes, the entire state is sent. While this doesn't appear to be a problem when synchronizing data from Global to Zone, it may be less efficient when there are many Dataplane changes. There are several improvements we want to make in the future such as:
+* working with different older versions of the control plane on the zone,
+* creating separate gRPC services for synchronization between the zone and global,
+* supporting multiple control plane tenants,
+* prioritizing sync of operational over informational resources,
+* changing `default` namespace for sync to global to system one (`kuma-system`), 
+* and sending only changes instead of the state of the world when synchronizing data from the zone to the global.
+
+## Considered Options
+
+* Redesign KDS
+
+## Decision Outcome
+
+Chosen option: "Redesign KDS".
+
+## Solution
+
+Currently, we are using one bidirection stream for sending resources from Zone to Global and Global to Zone.
+
+```proto
+service KumaDiscoveryService {
+ rpc StreamKumaResources(stream envoy.service.discovery.v3.DeltaDiscoveryRequest)
+     returns (stream envoy.service.discovery.v3.DeltaDiscoveryResponse);
+}
+```
+
+That is hidden behind `MultiplexService`.
+
+``` proto
+service MultiplexService {
+  rpc StreamMessage(stream Message) returns (stream Message);
+}
+```
+
+Zone to Global synchronize resources:
+
+* Dataplane
+* DataplaneInsight
+* ZoneEgress
+* ZoneEgressInsight
+* ZoneIngress*
+* ZoneIngressInsight
+
+`*` ZoneIngress resource is synchronized from Global to Zone and Zone to Global
+
+Global to Zone synchronize resources:
+* CircuitBreaker
```

Comment:
An exhaustive list is too noisy. Much more helpful for anyone reading this would be to say which _kind_ of resources are being synced G->Z and Z->G and why?

Reply:
changed it